engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/CallCache.scala

```diff
@@ -62,20 +69,31 @@ class CallCache(database: CallCachingSqlDatabase) {
       }
     }
 
-    CallCachingJoin(callCachingEntry, hashesToInsert.toSeq, resultToInsert.toSeq, jobDetritusToInsert.toSeq)
+    CallCachingJoin(callCachingEntry, hashesToInsert.toSeq, aggregatedHashesToInsert, resultToInsert.toSeq, jobDetritusToInsert.toSeq)
   }
 
-  def callCachingEntryIdsMatchingHashes(callCacheHashes: CallCacheHashes)(implicit ec: ExecutionContext): Future[Set[CallCachingEntryId]] = {
-    callCachingEntryIdsMatchingHashes(NonEmptyList.fromListUnsafe(callCacheHashes.hashes.toList))
+  def hasBaseAggregatedHashMatch(baseAggregatedHash: String)(implicit ec: ExecutionContext): Future[Boolean] = {
+    database.hasMatchingCallCachingEntriesForBaseAggregation(baseAggregatedHash)
+  }
+  
+  def hasKeyValuePairHashMatch(hashes: Set[HashResult])(implicit ec: ExecutionContext): Future[Boolean] = {
+    hashes.toList match {
+        // Empty hashes should match nothing
+      case Nil => Future.successful(false)
```

Comment:
Oh right yes ! I guess technically this would never need to be called with an empty set but still..